sds_data_manager/stacks/ialirt_processing_stack.py

```diff
@@ -46,78 +49,206 @@ def __init__(
+            # Add ingress rule for each port
+            ecs_security_group.add_ingress_rule(
+                peer=ec2.Peer.any_ipv4(),
+                connection=ec2.Port.tcp(port),
+                description="Allow inbound traffic on TCP port",
+            )
+        return ecs_security_group
+
+    def create_load_balancer_security_group(self, ports):
+        """Create and return a security group for load balancers."""
+        # Create a security group for the ALB
+        load_balancer_security_group = ec2.SecurityGroup(
             self,
-            "IalirtEC2SecurityGroup",
+            "ALBSecurityGroup",
             vpc=self.vpc,
-            description="Security group for Ialirt EC2 instance",
+            description="Security group for the Ialirt ALB",
         )
 
-        # Allow ingress to LASP IP address range and specific port
-        security_group.add_ingress_rule(
-            ec2.Peer.ipv4("128.138.131.0/24"),
-            ec2.Port.tcp(8080),
-            "Allow inbound traffic on TCP port 8080",
+        # Allow inbound traffic from a specific port and
+        # any ipv4 address.
+        for port in ports:
+            load_balancer_security_group.add_ingress_rule(
+                peer=ec2.Peer.any_ipv4(),
+                connection=ec2.Port.tcp(port),
+                description=f"Allow inbound traffic on " f"TCP port {port}",
+            )
+
+        # Allow all outbound traffic.
+        load_balancer_security_group.add_egress_rule(
+            peer=ec2.Peer.any_ipv4(),
+            connection=ec2.Port.all_traffic(),
+            description="Allow outbound traffic",
         )
 
-        # Create an IAM role for the EC2 instance
-        # - Read-only access to AWS ECR
-        # - Basic instance management via AWS Systems Manager
-        # Note: the Systems Manager provides a secure way for
-        # users to interact and access the EC2 during development.
-        ec2_role = iam.Role(
+        return load_balancer_security_group
+
+    def add_compute_resources(self, container_name, ports, ecs_security_group):
+        """Add ECS compute resources for a container."""
+        # ECS Cluster manages EC2 instances on which containers are deployed.
+        ecs_cluster = ecs.Cluster(self, f"IalirtCluster{container_name}", vpc=self.vpc)
+
+        # This task definition specifies the networking mode as AWS_VPC.
+        # ECS tasks in AWS_VPC mode can be registered with
+        # Application Load Balancers (ALB).
+        task_definition = ecs.Ec2TaskDefinition(
+            self, f"IalirtTaskDef{container_name}", network_mode=ecs.NetworkMode.AWS_VPC
+        )
+
+        # Adds a container to the ECS task definition
+        # Logging is configured to use AWS CloudWatch Logs.
+        container = task_definition.add_container(
+            f"IalirtContainer{container_name}",
+            image=ecs.ContainerImage.from_ecr_repository(self.repo, "latest"),
+            memory_limit_mib=512,
```

Comment:
Add a quick link to what these values mean?
```suggestion
            # Allowable values: 
            # https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecs.TaskDefinition.html#cpu
            memory_limit_mib=512,
```